### PR TITLE
Update the secure naming based on the SAN in the actual GCP certificates

### PR DESCRIPTION
**Pull request by lei-tang**
Update the secure naming based on the SAN in the actual GCP certificates

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/conversion.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func serviceHostname(name, namespace, domainSuffix string) model.Hostname {
 
 // canonicalToIstioServiceAccount converts a Canonical service account to an Istio service account
 func canonicalToIstioServiceAccount(saname string) string {
-	return fmt.Sprintf("%v://%v", IstioURIPrefix, saname)
+	return fmt.Sprintf("%v://accounts.google.com/%v", IstioURIPrefix, saname)
```
Comment (quanjielin) [inline pilot/pkg/serviceregistry/kube/conversion.go]: %s ?

Comment (lei-tang) [reply pilot/pkg/serviceregistry/kube/conversion.go]: Per the discussion, this PR keeps the original %v.

**Comment by wattli** [comment]
/hold

Please check in this code in branch collab-gcp-identity instead

**Comment by lei-tang** [comment]
This PR is moved to https://github.com/istio/istio/pull/7547 to commit the code into the collab-gcp-identity branch.

**Pull request closed without merging**